views/dataset/summary/summary.templ

```diff
@@ -5,25 +5,43 @@ import (
 	"net/url"
 	"github.com/ugent-library/biblio-backoffice/ctx"
 	"github.com/ugent-library/biblio-backoffice/models"
+	"github.com/ugent-library/biblio-backoffice/views"
 	"github.com/ugent-library/biblio-backoffice/views/contributor"
 	"github.com/ugent-library/biblio-backoffice/views/relatedorganization"
-	"github.com/ugent-library/biblio-backoffice/views"
 )
 
 type SummaryArgs struct {
 	Dataset *models.Dataset
 	URL     *url.URL
+	Target  string
 	Actions templ.Component
 	Footer  templ.Component
 	Links   templ.Component
 }
 
+func getUserContributorRoles(d *models.Dataset, user *models.Person) string {
+	if d.HasContributor("author", user) {
+		return "creator"
+	}
```

Comment:
This is a bit tricky though. If we ever rename dataset authors to creators (as is marked with a TODO), this could easily be forgotten (but a test would fail).

Reply:
@verheyenkoen It does make sense to add it as a separate view helper function because you'd end up with a complex block of conditionals in template otherwise. However, I think this function is a smell of two deeper issues we won't change on short notice.

1/ The `author` role to `creator` transformation is a band-aid to avoid a data transformation / code refactor. Regardless of where we put it, it's something to keep in mind later on anyway.

2/ I'd try making the case for model methods as a sort of "calculated / dynamic" properties for myself, but in our architectural design, I don't think that belongs at the model level such as it currently is. The `registrar` check drives that point  home: it's an example of why directly exposing an object from the data layer to the view layer creates inflexible tight coupling. Suppose we were to create a separate "view object" (or DTO) for a summary containing only relevant data to that specific view - instead of passing the entire publication / dataset - then we could add specific getter methods on that object and treat them as dynamic properties in the templ views. However, that's a significant change to our architecture, and something we need to think about first.

cc @nics

Reply:
this is all just about labels (a dataset author has as label "creator"), so it belongs in views and i18n

Reply:
@verheyenkoen just can remove the TODO about the field name, it won't happen and is not important

Reply:
@nics I'm not convinced it's just a label. It's not important right now, so let's discuss this later on.